src/Altinn.Apps/AppTemplates/AspNet/Altinn.App.Api/Controllers/DataController.cs

```diff
@@ -600,5 +613,89 @@ private static bool IsValidContributer(DataType dataType, ClaimsPrincipal user)
 
             return false;
         }
+
+        /// <summary>
+        /// Validated that the request 
+        /// </summary>
+        /// <param name="dataType"></param>
+        /// <param name="errorMessage"></param>
+        /// <returns></returns>
+        private bool CompliesWithDataRestrictions(DataType dataType, out string errorMessage)
+        {
+            errorMessage = string.Empty;
+
+            if (!Request.Headers.ContainsKey("Content-Disposition"))
+            {
+                errorMessage = "Conent-Disposition header containing 'filename' must be included in request.";
+                return false;
+            }
+
+            Request.Headers.TryGetValue("Content-Disposition", out StringValues headerValues);
+            string filename = GetFilenameFromHeader(headerValues.ToString());
+
+            if (string.IsNullOrEmpty(filename))
+            {
+                errorMessage = "Content-Disposition header must contain 'filename'.";
+                return false;
+            }
+
+            string[] splitFilename = filename.Split('.');
+
+            if (splitFilename.Count() != 2)
+            {
+                errorMessage = "Invalid format for filename. Expected format is {filename}.{filetype}.";
+                return false;
+            }
+
+            // no restrictions on data type
+            if (dataType.AllowedContentTypes == null || dataType.AllowedContentTypes.Count == 0)
+            {
+                return true;
+            }
+
+            string filetype = splitFilename[1];
+            string mimeType = MimeTypeMap.GetMimeType(filetype);
+
+            if (!Request.Headers.ContainsKey("Content-Type"))
+            {
+                errorMessage = "Content-Type header must be included in request.";
+                return false;
+            }
+
+            // Verify that file mime type matches content type in reuqest
+            Request.Headers.TryGetValue("Content-Type", out StringValues contentType);
+            if (!contentType.Equals("application/octet-stream") && !mimeType.Equals(contentType, StringComparison.InvariantCultureIgnoreCase))
+            {
+                errorMessage = $"Content type header {contentType} does not match mime type {mimeType} for uploaded file. Please fix header or upload another file.";
+                return false;
+            }
+
+            // Verify that file mime type is an allowed content-type
+            if (dataType.AllowedContentTypes != null && (!dataType.AllowedContentTypes.Contains(mimeType, StringComparer.InvariantCultureIgnoreCase) && !dataType.AllowedContentTypes.Contains("application/octet-stream")))
```

Comment:
Can simplify if expression. dataType.AllowedContentTypes != null is always true.

Reply:
Will it always be null thoguh? If app owner wan't to allow all content types I guess it could be empty, or?

Reply:
You are right, but logic a bit higher up also have a null check and the method returns if it is true. This means that down here it is never null.

Reply:
haha, look at that. I'd covered it all along. removing the check here.